src/main/java/org/jenkinsci/test/acceptance/plugins/git/GitScm.java

```diff
@@ -136,4 +198,199 @@ public RecursiveSubmodules(GitScm git, String path) {
+        }
+    }
+
+    public static class CustomNameAndMail extends Behaviour {
+        private final Control txtName = control("name");
+        private final Control txtEmail = control("email");
+
+        public CustomNameAndMail(GitScm git, String path) {
+            super(git, path);
+            clickLink("Custom user name/e-mail address");
+        }
+    }
+
+    public static class NoBuildOnCommit extends Behaviour {
+
+        public NoBuildOnCommit(GitScm git, String path) {
+            super(git, path);
+            clickLink("Don't trigger a build on commit notifications");
+        }
+    }
+
+    public static class ForcePollingUsingWorkspace extends Behaviour {
+
+        public ForcePollingUsingWorkspace(GitScm git, String path) {
+            super(git, path);
+            clickLink("Force polling using workspace");
+        }
+    }
+
+    public static class GitLfsPull extends Behaviour {
+
+        public GitLfsPull(GitScm git, String path) {
+            super(git, path);
+            clickLink("Git LFS pull after checkout");
+        }
+    }
+
+    public static class MergeBeforeBuild extends Behaviour {
+        private final Control txtMergeRemote = control("mergeRemote");
+        private final Control txtMergeTarget = control("mergeTarget");
+        private final Control selMergeStrategy = control("mergeStrategy");
+        private final Control selFastForwardMode = control("fastForwardMode");
+
+        public MergeBeforeBuild(GitScm git, String path) {
+            super(git, path);
+            clickLink("Merge before build");
+        }
+    }
+
+    public static class PollingIgnoresUser extends Behaviour {
+        private final Control txtExcludeUsers = control("excludeUsers");
+
+        public PollingIgnoresUser(GitScm git, String path) {
+            super(git, path);
+            clickLink("Polling ignores commits from certain users");
+        }
+    }
+
+    public static class PollingIgnoresPath extends Behaviour {
+        private final Control txtIncludedRegions = control("includedRegions");
+        private final Control txtExcludedRegions = control("excludedRegions");
+
+        public PollingIgnoresPath(GitScm git, String path) {
+            super(git, path);
+            clickLink("Pollinig ignores commits in certain paths");
+        }
+    }
+
+    public static class PollingIgnoresMessage extends Behaviour {
+        private final Control txtExcludedMessage = control("excludedMessage");
+
+        public PollingIgnoresMessage(GitScm git, String path) {
+            super(git, path);
+            clickLink("Polling ignores commits with certain messages");
+        }
+    }
+
+    public static class PruneStableRemoteBranches extends Behaviour {
+
+        public PruneStableRemoteBranches(GitScm git, String path) {
+            super(git, path);
+            clickLink("Prune stable remote-tracking branches");
+        }
+    }
+
+    // TODO Check if addPath actually works
```

Comment:
Instead of removing it, I decided to create another test which uses this class. Although it seemed to work, I rewrote .appPath() method to get rid of the ArrayList.